timelock-impl/src/test/java/com/palantir/atlasdb/timelock/batch/MultiClientConjureTimelockResourceTest.java

```diff
@@ -46,35 +51,58 @@
     private static final RedirectRetryTargeter TARGETER =
             RedirectRetryTargeter.create(LOCAL, ImmutableList.of(LOCAL, REMOTE));
 
-    private AsyncTimelockService timelockService = mock(AsyncTimelockService.class);
-    private LeaderTime leaderTime = mock(LeaderTime.class);
+    private Map<String, AsyncTimelockService> namespaces = new HashMap();
+    private Map<String, LeadershipId> namespaceToLeaderMap = new HashMap();
+
     private MultiClientConjureTimelockResource resource;
 
     @Before
     public void before() {
-        resource = new MultiClientConjureTimelockResource(TARGETER, unused -> timelockService);
+        resource = new MultiClientConjureTimelockResource(TARGETER, this::getServiceForClient);
     }
 
     @Test
     public void canGetLeaderTimesForMultipleClients() {
-        when(timelockService.leaderTime()).thenReturn(Futures.immediateFuture(leaderTime));
-        Set<Namespace> namespaces = ImmutableSet.of(Namespace.of("client1"), Namespace.of("client2"));
-        assertThat(Futures.getUnchecked(resource.leaderTimes(AUTH_HEADER, namespaces)))
-                .isEqualTo(getLeaderTimesForNamespaces(namespaces));
+        Namespace client1 = Namespace.of("client1");
+        Namespace client2 = Namespace.of("client2");
+        Set<Namespace> namespaces = ImmutableSet.of(client1, client1, client2);
```

Comment:
nit: just use one instance of `client1`? This won't really test multiple requests because the immutableset will kill off the doubles before we call resource.getLeaderTimes